openshift/templates/lup-api-mongodb-dev-deploy.yaml

```diff
@@ -0,0 +1,72 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: landuseplanning-api-mongodb-dev
+  namespace: e8b9ad-dev
+  labels:
+    app: landuseplanning-api-mongodb-dev
+    app.kubernetes.io/component: landuseplanning-api-mongodb-dev
+    app.kubernetes.io/instance: landuseplanning-api-mongodb-dev
+    app.kubernetes.io/part-of: landuseplanning-api-dev
+    app.openshift.io/runtime-namespace: openshift
+    app.openshift.io/runtime-version: latest
+spec:
+  strategy:
+    type: RollingUpdate
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: landuseplanning-api-mongodb-dev
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: landuseplanning-api-mongodb-dev
+        deploymentconfig: landuseplanning-api-mongodb-dev
```

Comment:
Were you having any trouble with the ReplicaSets and pods related to this Deployment? The Kubernetes docs seem to say that the `selector.matchLabels` and the `template.labels` need to be an exact match, otherwise the Deployment and ReplicaSet may have issues

Reply:
Yes, I was having issues with ReplicaSets, but it was unrelated to anything in this file. A deploymentConfig label had to be removed from the service. It appears that spec.selector.matchLabels.app and spec.template.metadata.labels.deploymentconfig are an exact match in this file.

Reply:
I saw in your review summary that it should be removed, so I removed it from the YAML

Reply:
Tried to restart rollout after removing the property and it incorrectly tried to create ReplicaSet again. Removed the property again and everything is working.